File: src/types/task.ts
export enum TaskStatus {
  Pending = 'Pending',
  Running = 'Running',
  Completed = 'Completed',
  Failed = 'Failed',
  Cancelled = 'Cancelled'
}

export type TaskType =
  | 'ImportSkill'
  | 'ScanSkill'
  | 'ScanRepository'
  | 'Download'
  | 'Other';

export enum ProgressStage {
  Queued = 'Queued',
  Preparing = 'Preparing',
  Downloading = 'Downloading',
  Scanning = 'Scanning',
  Analyzing = 'Analyzing',
  Installing = 'Installing',
  Finalizing = 'Finalizing',
  Completed = 'Completed',
  Failed = 'Failed',
  Cancelled = 'Cancelled'
}

export interface ProgressEvent {
  task_id: string;
  stage: ProgressStage | string;
  message: string;
  progress: number;
  percentage?: number;
  total?: number;
  current?: number;
}

export interface Task {
  id: string;
  task_type: TaskType;
  status: TaskStatus;
  title: string;
  progress?: ProgressEvent;
  created_at: string;
  started_at?: string;
  completed_at?: string;
  error?: string;
}

File: src-tauri/src/tasks/task.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};
use uuid::Uuid;
use super::progress::ProgressEvent;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum TaskType {
    ImportSkill,
    ScanSkill,
    ScanRepository,
    Download,
    Other(String),
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
pub enum TaskStatus {
    Pending,
    Running,
    Completed,
    Failed,
    Cancelled,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BackgroundTask {
    pub id: String,
    pub task_type: TaskType,
    pub status: TaskStatus,
    pub title: String,
    pub progress: Option<ProgressEvent>,
    pub created_at: DateTime<Utc>,
    pub started_at: Option<DateTime<Utc>>,
    pub completed_at: Option<DateTime<Utc>>,
    pub error: Option<String>,
}

impl BackgroundTask {
    pub fn new(task_type: TaskType, title: String) -> Self {
        Self {
            id: Uuid::new_v4().to_string(),
            task_type,
            status: TaskStatus::Pending,
            title,
            progress: None,
            created_at: Utc::now(),
            started_at: None,
            completed_at: None,
            error: None,
        }
    }
}
